src/main/java/org/mule/extension/ftp/internal/ftp/command/ClassicFtpCommand.java

```diff
@@ -176,4 +180,43 @@ public RuntimeException exception(String message, Exception cause) {
   private String enrichExceptionMessage(String message) {
     return format("%s. Ftp reply code: %d", message, client.getReplyCode());
   }
+
+  private Optional<FTPFile> getFileFromPath(Path path) throws IOException {
+    String filePath = path.toAbsolutePath().toString();
+    // Check if MLST command is supported
+    FTPFile file = client.mlistFile(filePath);
+    if (file == null) {
+      FTPFile[] files = client.listFiles(filePath);
+      if (files.length >= 1) {
+        if (filePath.endsWith(files[0].getName())) {
+          // List command result is the file from the path parameter
+          file = files[0];
+        } else {
+          // List command result is a directory
+          if (path.getParent() != null) {
+            files = client.listDirectories(path.getParent().toAbsolutePath().toString());
+            return Arrays.stream(files).filter(dir -> filePath.endsWith(dir.getName())).findFirst();
+          } else {
+            // root directory
+            file = createRootFile();
```

Comment:
you can't return this. This is an artificial, untruthful file. Doesn't contain real information about this root.